### PR TITLE
Run migrations with transactions disabled

### LINKED ISSUE #6: Crash when setting up default key during create
Excerpt from running `google-cloud-gemserver create` on a new project.

```
Saving configuration
Would you like to setup a default key? [Y/n] (default yes)

/usr/local/google/home/dazuma/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/google-cloud-gemserver-0.1.0/lib/google/cloud/gemserver/cli/server.rb:167:in `extract_key': undefined method `+' for nil:NilClass (NoMethodError)
        from /usr/local/google/home/dazuma/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/google-cloud-gemserver-0.1.0/lib/google/cloud/gemserver/cli/server.rb:107:in `setup_default_keys'
        from /usr/local/google/home/dazuma/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/google-cloud-gemserver-0.1.0/lib/google/cloud/gemserver/cli/server.rb:73:in `deploy'
        from /usr/local/google/home/dazuma/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/google-cloud-gemserver-0.1.0/lib/google/cloud/gemserver/cli.rb:68:in `create'
        from /usr/local/google/home/dazuma/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/thor-0.19.4/lib/thor/command.rb:27:in `run'
        from /usr/local/google/home/dazuma/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/thor-0.19.4/lib/thor/invocation.rb:126:in `invoke_command'
        from /usr/local/google/home/dazuma/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/thor-0.19.4/lib/thor.rb:369:in `dispatch'
        from /usr/local/google/home/dazuma/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/thor-0.19.4/lib/thor/base.rb:444:in `start'
        from /usr/local/google/home/dazuma/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/google-cloud-gemserver-0.1.0/lib/google/cloud/gemserver/cli.rb:43:in `start'
        from /usr/local/google/home/dazuma/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/google-cloud-gemserver-0.1.0/bin/google-cloud-gemserver:5:in `<top (required)>'
        from /usr/local/google/home/dazuma/.rbenv/versions/2.4.1/bin/google-cloud-gemserver:22:in `load'
        from /usr/local/google/home/dazuma/.rbenv/versions/2.4.1/bin/google-cloud-gemserver:22:in `<main>'
```

### PR DESCRIPTION
**Pull request by arhamahmed**
Should fix [this](https://github.com/GoogleCloudPlatform/google-cloud-gemserver/issues/6) bug where having `use_transactions: true` resulted in `Mysql2::Error: SAVEPOINT autopoint_1 does not exist` due to implicit commit when `Gemstash::Env.migrate` ran.

Database connections are now properly cached, as well.

### CODE REVIEW

**Review by hxiong388 — APPROVED** [review]

File: lib/patched/env.rb [hunk 1]
```diff
@@ -20,14 +20,29 @@ module PatchedEnv
   ##
   # Monkey patch to support Cloud SQL as an adapter
   def db
+    db = instance_variable_get :@db
+    return db if db
```
Comment (hxiong388) [inline lib/patched/env.rb]: These two lines can now be simplified as `return @db if @db`. The set statement below can just set `@db` by using `@db=`.

Comment (arhamahmed) [reply lib/patched/env.rb]: Hmm I see.

File: lib/google/cloud/gemserver/cli/request.rb [hunk 2]
```diff
@@ -40,7 +40,7 @@ class Request
           # gemserver.
           def initialize url = nil
             gemserver_url = url.nil? == true ? remote : url
-            @http = Net::HTTP.new gemserver_url
+            @http = Net::HTTP.new gemserver_url, 8080
```
Comment (dazuma) [inline lib/google/cloud/gemserver/cli/request.rb]: Why port 8080? I'd assume it should normally go through port 80 as usual.

Comment (arhamahmed) [reply lib/google/cloud/gemserver/cli/request.rb]: Server runs on port 8080.

**Review by dazuma — APPROVED** [review]

**Pull request merged**